Python/bytecodes.c

```diff
@@ -2978,36 +2978,39 @@ dummy_func(
             CHECK_EVAL_BREAKER();
         }
 
-        // error: MAKE_FUNCTION has irregular stack effect
-        inst(MAKE_FUNCTION) {
-            PyObject *codeobj = POP();
-            PyFunctionObject *func = (PyFunctionObject *)
+        inst(MAKE_FUNCTION, (defaults    if (oparg & 0x01),
+                             kwdefaults  if (oparg & 0x02),
+                             annotations if (oparg & 0x04),
+                             closure     if (oparg & 0x08),
+                             codeobj -- func)) {
```

Comment:
Too bad we can't write `func: PyFunctionObject *` yet -- I've gotta add that to the grammar, it would add the casts and save the ugly extra `func_obj` variable.